packages/api/src/controllers/helpers.ts

```diff
@@ -234,6 +237,35 @@ export async function getS3PresignedUrl(os: ObjectStore, objectKey: string) {
   return getSignedUrl(s3, putCommand, { expiresIn });
 }
 
+export async function generateRequesterId(req: Request, playbackId: string) {
+  const origin =
+    req.headers["cf-connecting-ip"] ||
+    req.headers["true-client-ip"] ||
+    req.headers["x-forwarded-for"];
+
+  let requesterId: string;
+  if (!origin) {
+    console.log(`
+        unable to determine origin of requester for user=${req.user.id} when clipping playbackId=${playbackId}
+    `);
+    requesterId = `UNKNOWN-${playbackId}`;
+  } else {
+    console.log(`
+         user=${req.user.id} requesterId generated for playbackId=${playbackId} from origin=${origin}
+    `);
+    let originString = Array.isArray(origin) ? origin.join(",") : origin;
```

Comment:
another simpler alternative is to just make `ip.toString()` whenever you use it. if it's already a string, the same string is returned